backend/fms_core/tests/test_services/sample_next_step_test.py

```diff
@@ -292,6 +292,61 @@ def test_move_sample_to_next_step(self):
         self.assertEqual(StepHistory.objects.filter(study=study_B, process_measurement__source_sample=sample_in, step_order__step=step).count(), 1)
         self.assertEqual(StepHistory.objects.filter(study=study_C, process_measurement__source_sample=sample_in, step_order__step=step).count(), 1)
 
+    def test_move_sample_to_next_step_repeat_step(self):
+        container1 = Container.objects.create(**create_sample_container(kind='tube', name='TestTube01_1', barcode='T123456_1'))
+        sample_in = create_fullsample(name="TestSampleNextStep_in",
+                                      alias="TestSampleNextStep_in",
+                                      volume=5000,
+                                      individual=self.valid_individual,
+                                      sample_kind=self.sample_kind_DNA,
+                                      container=container1,
+                                      project=self.project)
+        
+        step = Step.objects.get(name="Normalization (Sample)")
+        for derived_by_sample in sample_in.derived_by_samples.all():
+            derived_by_sample.project_id = self.project.id
+            derived_by_sample.save()
+                
+        letter_B = "B"
+        start = 3
+        end = 7
+        study_B = Study.objects.create(letter=letter_B,
+                                       project=self.project,
+                                       workflow=self.workflow_pcr_free,
+                                       start=start,
+                                       end=end)
+
+        letter_C = "C"
+        start = 3
+        end = 7
+        study_C = Study.objects.create(letter=letter_C,
+                                       project=self.project,
+                                       workflow=self.workflow_pcr_enriched,
+                                       start=start,
+                                       end=end)
+
+        protocol = Protocol.objects.get(name="Normalization")
+        process = Process.objects.create(protocol=protocol)
+        process_measurement = ProcessMeasurement.objects.create(process=process,
+                                                                source_sample=sample_in,
+                                                                execution_date=datetime.date(2021, 1, 10),
+                                                                volume_used=10)
+
+        old_sample_to_study_workflow_1, _, _ = queue_sample_to_study_workflow(sample_in, study_B)
+        old_sample_to_study_workflow_2, _, _ = queue_sample_to_study_workflow(sample_in, study_C)
+        new_sample_next_steps, errors, warnings = move_sample_to_next_step(step, sample_in, process_measurement, WorkflowAction.REPEAT_STEP, keep_current=True)
```

Comment:
Still kinda weird that a sample sample can be in two different studies and be moved to the next at the same time in two different studies

Reply:
you could want to do more than one thing with a sample ...